### PR TITLE
test/random_failures: fix gossip shadow round injection

### PR DESCRIPTION
**Pull request by andrzej-jackowski-scylladb**
Commit c17c4806a1f0c423dfc276b4419eec2775e570b1 removed check_for_endpoint_collision() from
the fresh bootstrap path, which was the only code path that
called do_shadow_round() for new nodes. Since the gossip shadow
round is no longer executed during bootstrap, remove the
stop_during_gossip_shadow_round error injection from the test.

The entry is marked as REMOVED_ rather than deleted to preserve
the shuffle order for seed-based test reproducibility.

The injection point in gms/gossiper.cc is also removed since it
is no longer used by any test.

Fixes: SCYLLADB-1466

No backport, as the fix is needed only on master

### CODE REVIEW

File: test/cluster/random_failures/test_random_failures.py [hunk 1]
```diff
@@ -135,6 +135,24 @@ async def test_random_failures(manager: ManagerClient,
             node_ip=coordinator.ip_addr,
             injection="topology_coordinator_pause_after_updating_cdc_generation",
         )
+    elif error_injection == "stop_during_gossip_shadow_round":
```
Comment (patjed41) [inline test/cluster/random_failures/test_random_failures.py]: I think the goal of this test was to test random failures during bootstrap. Your change is against that. 

Can't we just remove the injection? It makes sense to me -- the test is for bootstrap, but there is no shadow round during bootstrap anymore.

File: test/cluster/random_failures/test_random_failures.py
Comment (patjed41) [inline test/cluster/random_failures/test_random_failures.py:133]: > After that change, the gossip shadow round is only executed on restart (bootstrap_complete() == true) or replacement (is_replacing()).

@gleb-cloudius Is that right? It's suspicious that we perform shadow round for replace but not for bootstrap.

File: test/cluster/random_failures/test_random_failures.py (hunk 1 above)
Comment (andrzej-jackowski-scylladb) [reply test/cluster/random_failures/test_random_failures.py]: @patjed41, I think you are right that the goal of this test is to verify random failures during bootstrap, and removing the error injection was my first idea.                                                                                   

However, `stop_during_gossip_shadow_round` is not used in any other test, and it seems to me that this path should be covered somewhere. I didn't want to add a new test, so I added this `elif` to keep the coverage as close to the original as possible. There is already a similar hack to handle `stop_after_updating_cdc_generation` error injection.

If you insist on removing it, or @gleb-cloudius confirms that testing this path makes little sense, I will do it. 

I can also add a new `random_failures` test for restarts, but I'm concerned it will be flaky in some corner-cases, and my primary goal was to increase the CI stability.

**Comment by andrzej-jackowski-scylladb** [comment]
> #### 🔴 CI State: FAILURE
> Mode	Stage	Status	Queue Time	Run Time	Node
> [Checkout](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/28295//pipeline-overview)	✅		2m	spider9.cloudius-systems.com
> [Framework test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/28295//pipeline-overview)	✅		0m	spider9.cloudius-systems.com
> [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/28295//pipeline-overview)	✅	0m	28m	spider16.cloudius-systems.com
> debug	[Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/28295//pipeline-overview)
> 🔹 test_random_failures	❌	0m	8m	spider16.cloudius-systems.com
> #### Build Details:
> * Total Queue: 0m
> * Total Run: 39m
> 
> **Note:** To re-trigger CI for this PR, comment: `@scylladbbot trigger-ci`

Based on the logs all tests passed:
```
14:51:39  bringing up nodes...
14:51:40  bringing up nodes...
14:51:40  
14:56:56  ........................................................................ [ 72%]
14:59:01  ............................                                             [100%]
14:59:01  -- generated xml file: /scylladir/testlog/x86_64/report/pytest_cpp_d509f.xml ---
14:59:01  100 passed in 442.78s (0:07:22)
14:59:01  Driver name ScyllaDB Python Driver, version 3.29.7
14:59:01  ================================================================================
14:59:01  [N/TOTAL]   SUITE    MODE   RESULT   TEST
14:59:01  ------------------------------------------------------------------------------
14:59:01  
14:59:01  ------------------------------------------------------------------------------
14:59:02  CPU utilization: 64.0%
```

I think the failure is caused by a temporary CI problems. I will re-try when the environment problems are solved.

File: test/cluster/random_failures/test_random_failures.py
Comment (gleb-cloudius) [reply test/cluster/random_failures/test_random_failures.py:133]: It is. Not only that, but I wanted to remove it from restart as well because the only reason it is there is to check snitch name mismatch and  according to support this is incorrect behaviour, snitch names should not matter. Shadow round during replace is required since the joining node needs to get an information about the node it replaces (like its dc/rack for instance or if replace by IP is used then we need to find out id we replace). If we will finally drop replace by IP support I think we can move dc/rack check to the join requests handling and get rig of shadow round all together.

File: test/cluster/random_failures/test_random_failures.py (hunk 1 above)
Comment (gleb-cloudius) [reply test/cluster/random_failures/test_random_failures.py]: Yes, lets just remove this particular random injection point.

Comment (patjed41) [reply test/cluster/random_failures/test_random_failures.py]: It's true that there already is a hack for `stop_after_updating_cdc_generation`, but with that injection the node is still bootstrapping, not restarting.

I agree that adding a new test would be an overkill. I also think we don't have to keep the injection exactly as it is. I don't see why testing a crash during the shadow round is particularly interesting. Moving the injection to the code just before or just after the shadow round shouldn't change anything from the topology changes' point of view. So, I believe we can just move this injection to the beginning of `storage_service::join_topology` executed in all paths (bootstrap, replace, restart), and rename it appropriately.

EDIT: I didn't see the Gleb's comment above.

**Comment by andrzej-jackowski-scylladb** [comment]
v2:
 - Removed `stop_during_gossip_shadow_round` as discussed in https://github.com/scylladb/scylladb/pull/29460#discussion_r3073450017

File: test/cluster/random_failures/test_random_failures.py (hunk 1 above)
Comment (andrzej-jackowski-scylladb) [reply test/cluster/random_failures/test_random_failures.py]: Ok, I removed `stop_during_gossip_shadow_round` from the test and from cpp (because now it's not used anywhere).

**Comment by andrzej-jackowski-scylladb** [comment]
v3:
 - Removed the `stop_during_gossip_shadow_round` injected block from cpp as well

**Comment by andrzej-jackowski-scylladb** [comment]
> #### 🔴 CI State: FAILURE
> Mode	Stage	Status	Queue Time	Run Time	Node
> [Checkout](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/28323//pipeline-overview)	✅		2m	spider9.cloudius-systems.com
> [Framework test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/28323//pipeline-overview)	✅		0m	spider9.cloudius-systems.com
> [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/28323//pipeline-overview)	✅	32m	22m	spider14.cloudius-systems.com
> dev	[test.py](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/28323//pipeline-overview)	❌		15m	spider14.cloudius-systems.com
> #### Failed Tests (1/8616):
> * [test_row_ttl_scheduling_group.dev.1](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/28323/artifact/testlog/x86_64/pytest_tests_logs/cluster-test_ttl_row.py-test_row_ttl_scheduling_group.dev.1-call-bdbac.log) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%22test_row_ttl_scheduling_group.dev.1%22) [🎫](https://scylladb.atlassian.net/issues/?jql=text~%22test_row_ttl_scheduling_group%22) [📂](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/28323/artifact/testlog/x86_64/dev/failed_test/test_row_ttl_scheduling_group.dev.1/)
> 
> 📄 Elasticsearch analysis for **test_row_ttl_scheduling_group**
> Period	CI Failures / Total	CI Pass Rate (%)	Next Failures / Total	Next Pass Rate (%)
> Week (7d)	15/1063	98.59	2/597	99.66
> Month (30d)	16/4005	99.6	2/2187	99.91
> Quarter (90d)	17/6203	99.73	2/3303	99.94
> #### Recent 10 Failures
> Test Name	Arch	Mode	Jenkins Node	Jenkins Job	Ref	Timestamp	Days Ago
> test_row_ttl_scheduling_group.dev.1	x86_64	dev	spider14.cloudius-systems.com	scylla-master/scylla-ci	PR#29460	2026-04-13 19:02	today
> test_row_ttl_scheduling_group.release.1	x86_64	release	spider18.cloudius-systems.com	scylla-master/scylla-ci	PR#29347	2026-04-13 18:20	today
> test_row_ttl_scheduling_group.release.2	x86_64	release	spider18.cloudius-systems.com	scylla-master/scylla-ci	PR#29347	2026-04-13 18:20	today
> test_row_ttl_scheduling_group.dev.1	x86_64	dev	spider15.cloudius-systems.com	scylla-master/scylla-ci	PR#29463	2026-04-13 17:59	today
> test_row_ttl_scheduling_group.dev.1	x86_64	dev	spider14.cloudius-systems.com	scylla-master/scylla-ci	PR#29461	2026-04-13 15:13	today
> test_row_ttl_scheduling_group.dev.1	x86_64	dev	spider14.cloudius-systems.com	scylla-master/scylla-ci	PR#29403	2026-04-13 14:38	today
> test_row_ttl_scheduling_group.dev.1	x86_64	dev	spider15.cloudius-systems.com	scylla-master/scylla-ci	PR#29457	2026-04-13 14:30	today
> test_row_ttl_scheduling_group.dev.1	x86_64	dev	spider12.cloudius-systems.com	scylla-master/scylla-ci	PR#29305	2026-04-13 13:43	today
> test_row_ttl_scheduling_group.dev.79	x86_64	dev	spider17.cloudius-systems.com	scylla-master/scylla-ci	PR#29447	2026-04-13 12:28	today
> test_row_ttl_scheduling_group.dev.71	x86_64	dev	spider17.cloudius-systems.com	scylla-master/scylla-ci	PR#29447	2026-04-13 12:28	today
> #### Build Details:
> * Total Queue: 32m
> * Total Run: 1h 11m
> 
> **Note:** To re-trigger CI for this PR, comment: `@scylladbbot trigger-ci`

SCYLLADB-1495

**Comment by andrzej-jackowski-scylladb** [comment]
@scylladbbot trigger-ci

**Review by patjed41 — APPROVED** [review]

**Comment by patjed41** [comment]
^ again https://scylladb.atlassian.net/browse/SCYLLADB-1495

@scylladbbot trigger-ci

**Review by gleb-cloudius — APPROVED** [review]

**Comment by andrzej-jackowski-scylladb** [comment]
@scylladb/scylla-maint please consider merging

**Comment by denesb** [comment]
Need green CI.

**Comment by andrzej-jackowski-scylladb** [comment]
> #### Failed Tests (1/62914):
> * [test_random_failures[stop_before_streaming-kill_coordinator_node].debug.3](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/28340/artifact/testlog/x86_64/pytest_tests_logs/cluster-random_failures-test_random_failures.py-test_random_failures%5Bstop_before_streaming-kill_coordinator_node%5D.debug.3-call-3796d.log) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+%22test_random_failures%5Bstop_before_streaming-kill_coordinator_node%5D.debug.3%22) [🎫](https://scylladb.atlassian.net/issues/?jql=text~%22test_random_failures%22) [📂](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/28340/artifact/testlog/x86_64/debug/failed_test/test_random_failures%5Bstop_before_streaming-kill_coordinator_node%5D.debug.3/)
> 
> 📄 Elasticsearch analysis for **test_random_failures[stop_before_streaming-kill_coordinator_node]**

Reported SCYLLADB-1553. It's a failure of the same test but it's unrelated to this PR.

**Comment by andrzej-jackowski-scylladb** [comment]
@scylladbbot trigger-ci
